homeassistant/components/generic_thermostat/config_flow.py

```diff
@@ -69,6 +80,36 @@
             mode=selector.NumberSelectorMode.BOX, unit_of_measurement=DEGREE, step=0.1
         )
     ),
+    vol.Optional(CONF_PRECISION): vol.All(
```

Comment:
So, we want to limit the number of options available in the UI. So these are not exposed on purpose.

Besides, introducing a field that has multiple meanings (select + float) is something we shouldn't do, IMHO.

Reply:
>  introducing a field that has multiple meanings (select + float) is something we shouldn't do, IMHO.

I was torn here between how the current config works:
* In reality, it's an enum, because there are only three allowed options (0.1, 0.5, 1)
* But it's modelled/stored as a float

So I was roundtripping this to a string in the UI (required by the select selector, to get a dropdown), and back to a float (to match the config format).

This could be simplified if the config was allowed to store a string representation.